supabase/migrations/083_profile_setup_onboarding.sql

```
@@ -0,0 +1,94 @@
+-- First-login onboarding: track whether a user has completed initial profile setup.
+-- External OAuth (Google/Discord) users start with random usernames and setup_completed=false,
+-- so they are guided to the onboarding screen. Email/password registrants pick their own
+-- username explicitly, so they start completed. Existing users are backfilled as completed.
+
+ALTER TABLE public.profiles
+    ADD COLUMN IF NOT EXISTS setup_completed boolean NOT NULL DEFAULT false;
+
+-- Existing accounts already have chosen names; don't force them through onboarding.
+UPDATE public.profiles SET setup_completed = true WHERE setup_completed = false;
```

SQLFluff (4.2.2)

[error] The 'WHERE' keyword should always start a new line.

(LT14)